crates/cheatnet/src/state.rs

```diff
@@ -88,6 +89,8 @@ pub struct CheatcodeState {
     pub pranked_contracts: HashMap<ContractAddress, ContractAddress>,
     pub warped_contracts: HashMap<ContractAddress, Felt252>,
     pub mocked_functions: HashMap<ContractAddress, HashMap<EntryPointSelector, Vec<StarkFelt>>>,
+    pub spies: Vec<SpyOn>,
```

Comment:
```suggestion
    pub event_spy_targets: Vec<SpyOn>,
```

Reply:
There is a problem with this approach.
Currently, we can assign an id to the `EventSpy` while creating it with the `spy_events` and then return the wanted events. We have to somehow know which spy wants which events. But it could be nice optimisation 🤔